plugins/discourse-local-dates/assets/javascripts/lib/discourse-markdown/discourse-local-dates.js

```diff
@@ -127,6 +100,78 @@ function addLocalDate(buffer, matches, state) {
   closeBuffer(buffer, state, dateTime.utc().format(config.format));
 }
 
+function defaultDateConfig() {
+  return {
+    date: null,
+    time: null,
+    timezone: null,
+    format: null,
+    timezones: null,
+    displayedTimezone: null,
+    countdown: null,
+    range: false,
+  };
+}
+
+function addLocalDate(buffer, matches, state) {
+  let config = defaultDateConfig();
+
+  const matchString = matches[1].replace(/‘|’|„|“|«|»|”/g, '"');
+
+  const parsed = parseBBCodeTag(
+    "[date date" + matchString + "]",
+    0,
+    matchString.length + 11
+  );
+
+  config.date = parsed.attrs.date;
+  config.format = parsed.attrs.format;
+  config.calendar = parsed.attrs.calendar;
+  config.time = parsed.attrs.time;
+  config.timezone = (parsed.attrs.timezone || "").trim();
+  config.recurring = parsed.attrs.recurring;
+  config.timezones = parsed.attrs.timezones;
+  config.displayedTimezone = parsed.attrs.displayedTimezone;
+  config.countdown = parsed.attrs.countdown;
+  addSingleLocalDate(buffer, state, config);
+}
+
+function addLocalRange(buffer, matches, state) {
+  let config = defaultDateConfig();
+  let date, time;
+  const matchString = matches[1].replace(/‘|’|„|“|«|»|”/g, '"');
```

Comment:
Maybe this should be a constant or function since it's used in two different places?

Reply:
Sounds good, I extracted it to `parseTagAttributes` function